2D Wave solver - Mesh parser, mesh plotter, isoparametric mapping, and sphinx conf.

amanabt opened the pull request:
This is the beginning of the development of the 2D wave equation solver. This PR I will:

1. Write the README following the Quazartech style guidelines.
2. Write a `params` file which will contain the user inputs required to run the program.
3. Storing of 2<sup>nd</sup> order quadrangular mesh.
4. Write a function to display the mesh grid.
5. Write a function to read the 2<sup>nd</sup> order quadrangular `.msh` files.
6. Configure `sphinx` for this project.
6. Write function to map an element of the mesh to
![image](https://user-images.githubusercontent.com/14011262/30854025-1392aa1e-a2ce-11e7-9695-730da9e3ba64.png) coordinates, where,
![image](https://user-images.githubusercontent.com/14011262/30854001-01151750-a2ce-11e7-837b-2dcf565414d1.png)
![image](https://user-images.githubusercontent.com/14011262/30853987-f4692b90-a2cd-11e7-87d0-61c367e77150.png).
This method of mapping is called isoparametric mapping.

amanabt committed:
Create LICENSE

amanabt committed:
Comany information, authors added.

amanabt committed:
.gitignore updated and README added.

amanabt committed:
Update README.md

amanabt committed:
Added "!.svgs/*" to .gitignore.
Added 2d_wave_eqn.png as needed for README.

amanabt committed:
Added 2d_wave_eqn.svg

amanabt committed:
.svgs/2d_wave_eqn resized.

amanabt committed:
svgs for F, c and u added.

amanabt committed:
Minor changes to 2d_wave_eqn svg

amanabt committed:
x_y.svg added.

amanabt committed:
Introduction to the project added.

amanabt committed:
svgs for README added.

amanabt committed:
Merge branch '2d_wave_solver' of github.com:amanabt/DG_Maxwell into 2d_wave_solver

amanabt committed:
hat_i and hat_j svgs added.

amanabt committed:
Minor changes in the README

- Character length of each line reduced.
- Minor addition to description of c_0 and c_1.

amanabt committed:
Maintainers added in README.

amanabt committed:
Shinx setup and lagrange.py and utils.py added.
- Sphinx documentation is setup but further work needs to be done to make it more readable.
- In lagrange.py, only LGL_points function is added.
- Sphinx gitignore added.

amanabt committed:
minor changes in index.rst. Added utils.rst and lagrange.rst to the documentation.

amanabt committed:
Submodules section added in the documentation.

amanabt committed:
doc/README.rst added.

amanabt committed:
Documentation README added and populated.

amanabt committed:
Renamed README.rst to README

amanabt committed:
README renamed to README.rst

amanabt committed:
Renamed README.md as README.rst

amanabt committed:
README.rst renamed to README.md

amanabt committed:
README.rst renamed to README.md

Syntax changed from rst to md.

amanabt committed:
Merge pull request #30 from QuazarTech/add-license-1

GPL-v3 license.

amanabt committed:
Mesh parser implemented.
- Documentation added for sphinx.
- Documentation checked in sphinx.
- Realised that I am using python2-sphinx to generate documentation which is causing
because of the code being written in python3.

amanabt committed:
Installation steps for sphinx and numpydoc added.

amanabt committed:
Isoparamatric mapping function added.
- Isoparam is currently using numpy array.
- plot_grid function added. Not working as expected.
- Functions are not documented.
- Files moved from lib directory to 2d_solver directory.

amanabt committed:
plot_element_grid function added, tested to be working properly.
- Documentation for plot_element_grid function done.
- Minor changes in sphinx files.

amanabt committed:
Isoparam functions documented.
Other minor documentation related changes.

amanabt committed:
Functions to plot lines, element boundaries, and mesh grid implemented.
* Documentation done for all the functions.
* All the functions tested.
* Unit tests not written.

amanabt committed:
Update README.md

amanabt committed:
Update README.md

amanabt committed:
unit test for isoparam_x added.

amanabt committed:
Minor changes in test/isoparam

amanabt committed:
Unit test for read_order_2_msh function added and tested.
- Unit test documented.

amanabt committed:
Written unit test for the function LGL_points.
- Unit test documented.
- All unit tests passing.

Removed unused functions from utils.py

amanabt committed:
Examples written, read_and_plot_mesh and plot_line

amanabt committed:
Update README.md

amanabt committed:
Update README.md

amanabt committed:
Update README.md

amanabt committed:
Update README.md

Balavarun5 commented on 2d_solver/isoparam.py:
```diff
@@ -0,0 +1,97 @@
+
+def isoparam_x(x_nodes, xi, eta):
+    '''
+    This function allows isoparametric mapping of a :math:`2^{nd}` order
```
The usage of 'function' can be avoided

amanabt replied:
done

Balavarun5 commented on 2d_solver/msh_parser.py:
```diff
@@ -0,0 +1,300 @@
+    
+    def is_9_node_quadrangle (eletag, eletype,
+                              physgrp, nodes):
+        return eletype == msh_handler.quadrangle_9_node
```
A variable can be returned instead of an expression

amanabt replied:
these are one line functions. I don't think that we need to complicate them. The name of the function is descriptive enough.

Balavarun5 commented on 2d_solver/msh_parser.py:
```diff
@@ -0,0 +1,300 @@
+
+    def save_element (eletag, eletype,
+                      physgrp, node_tags):
+        
```
Documentation is missing

amanabt replied:
Name of the function is already descriptive.

Balavarun5 commented on 2d_solver/msh_parser.py:
```diff
@@ -0,0 +1,300 @@
+    xi_LGL  : np.array [N_LGL]
+              LGL points on the :math:`\\xi` axis
+
+    eta_LGL  : np.array [N_LGL]
```
Alignment of the colons

amanabt replied:
checked at all other places and modified the documentations accordingly.

amanabt committed:
Codacy badge added.

amanabt committed:
minor changes

mchandra commented:
Does this PR contain all of the code required for 2D wave eqn evolution? Or only isometric mapping and A matrix assembly?

amanabt committed:
Minor corrections to code style according to Codacy suggestions.

amanabt commented:
A list of what the PR contains is written in the PR description. It does not contain code for A matrix assembly. I thought that it would be better to give a separate PR for A matrix assembly.

Ok, I think that the current title is not the appropriate gist of the PR. I will change it.

amanabt committed:
Minor changes

amanabt committed:
Minor changes in code style

amanabt committed:
Minor changes

amanabt committed:
Arrayfire removed from params.py

amanabt committed:
2d_solver renamed to solver_2d_wave because python import does not work with names starting with numericals.

amanabt committed:
2d_solver renamed to solver_2d_wave
- changes were made accordingly in docs/conf.py, examples/plot_line.ipynb, examples/read_and_plot_mesh/read_and_plot_mesh.ipynb.
- .travis.yml updated.
- requirements.txt added.

amanabt committed:
Minor change in .travis.yml

amanabt committed:
Update README.md

amanabt committed:
2D_wave_solver merger to master

amanabt committed:
Merge 2D_wave_solver and quazartech:DG_Maxwell:master.

amanabt committed:
svgs for README added.

amanabt committed:
Update README.md

- Integrated 1D wave equation and 2D wave equation README.

amanabt committed:
Removing solver_2d_wave directory

amanabt committed:
Documentation changes, unit test functions changes.
- Minor changes in documentation where reST syntax was not being followed.
- Minor changes in the unit test codes.
- All unit tests are passing.

amanabt committed:
Minor changes in the documentation.
- Added wave_equation.rst.
- Added wave_equation in submodules.rst.
- Removed instances of `References` being used in the documentation.
  Because this keyword is not recognisable by numpydoc and it it's there
  as a section, then numpydoc just gives a warning and does not process
  whatever is under this section.

amanabt committed:
Code Style changes according to CODACY suggestions.

amanabt committed:
Minor change in .travis.yml

amanabt committed:
Minor changes.

amanabt committed:
Arrayfire installation added in .travis.yml

amanabt committed:
minor changes in .travis.yml

amanabt committed:
minor changes in .travis.yml

amanabt committed:
minor changes in .travis.yml

amanabt committed:
Renamed directory 'code' to 'dg_maxwell' and bugfix in text_dx_dxi
- In test_dx_dxi, on keeping backend, the absolute value was not being taken
  when checking the difference between analytical and numerical dx_dxi.
  Because of this, the difference would come out to be +ve when using
  Arrayfire CPU backed and would come out to be -ve when using Arrayfire
  CUDA backend. Hence failing and passing the test just on changing the
  backend. This issue is fixed.
- Minor changes were made in the docs following the renaming of the `code`
  directory.

amanabt committed:
Minor changes in examples on following the renaming of `code` cirectory to `dg_maxwell` directory.

amanabt committed:
pytest added to .travis.yml script

amanabt committed:
pytest removed from .travis.yml
